web_widget_heatmap/static/src/js/heatmap_widget.js

```diff
@@ -0,0 +1,194 @@
+            });
+            var controls = false;
+            if (max_date < end_date) {
+                controls = true;
+            }
+            return {
+                start: start_date,
+                data: _.chain(timestamps).countBy().value(),
+                dataType: 'json',
+                minDate: start_date,
+                maxDate: max_date,
+                controls: controls,
+                range: range,
+                domain: domain,
+                subDomain: "hour",
+                domainGutter: 0,
+                highlight: "now",
+                onClick: this.onClickHeatMap.bind(this),
+                label: {
+                    position: "top"
+                }
+            };
+        },
+
+        /**
+         * Render the view
+         * @private
+         */
+        _render: function () {
+            var self = this;
+            var elements = this.value ? _.pluck(this.value.data, 'data') : [];
+            var options = this.generate_element_options(elements);
+            this.controls = options.controls;
+            this.renderElement();
+            var nodeOptions = this.nodeOptions || {};
+            this.heatmap_options = _.extend(options, {itemSelector: this.$el.find('.o_field_heatmap')[0]});
+            _.each(nodeOptions, function(value, key) {
+                self.heatmap_options[key] = value;
+            });
+            this.heatmap = new CalHeatMap();
+            this.heatmap.init(this.heatmap_options);
+        },
+
+        /**
+         * Renders the element and add new events.
+         */
+        renderElement: function() {
+            this._super();
+            this.$el.find('.next').click(this.nextHeatMap.bind(this));
+            this.$el.find('.previous').click(this.previousHeatMap.bind(this));
+        },
+
+        /**
+         * Show next heatmap grid
+         */
+        nextHeatMap: function() {
+            this.heatmap.next();
+        },
+
+        /**
+         * Show previous heatmap grid
+         */
+        previousHeatMap: function() {
+            this.heatmap.previous();
+        },
+
+        /**
+         * Find the records that match the selected date and show in the tree view
+         * @param {Date} date
+         * @returns {jQuery.Deferred} Action loaded
+         */
+        onClickHeatMap: function(date) {
+            date = moment(date);
+            var options = this.heatmap_options;
+            var elements = this.value ? _.pluck(this.value.data, 'data') : [];
+            if (elements && elements.length) {
+                // TODO: make faster
+                var current_elements_ids = elements.filter(function(el) {
+                    var duration = moment.duration(date.diff(el.create_date));
+                    if (options.domain === 'hour') {
+                        return duration.years() === 0 && duration.months() === 0 && duration.weeks() === 0 && duration.days() === 0 && duration.hours() === 0;
```

Comment:
Can be better for not equal cases change the order:
```suggestion
                        return duration.hours() === 0 && duration.days() === 0 && duration.weeks() === 0 && duration.months() === 0 && duration.years() === 0;
```